app/gui/view/component-browser/component-group/src/lib.rs

```diff
@@ -180,10 +182,35 @@ impl component::Frp<Model> for Frp {
         }
 
 
-        // === Header ===
+        // === Colors ===
 
+        let app_bg_color = style.get_color(ensogl_hardcoded_theme::application::background);
+        let header_intensity = style.get_number(theme::header::text::color_intensity);
+        let background_intensity = style.get_number(theme::background_color_intensity);
+        let dimmed_intensity = style.get_number(theme::dimmed_color_intensity);
+        let entry_text_color = style.get_color(theme::entries::text::color);
         frp::extend! { network
             init <- source_();
+            one <- init.constant(1.0);
+            intensity <- input.set_dimmed.switch(&one, &dimmed_intensity);
+            main_color <- all_with3(&app_bg_color, &input.set_color, &intensity,
+                |a,b,c| color::mix(*a,*b,*c));
```

Comment:
You can define a local `mix` function that would do it and use it here instead. Then a lot of lines will turn into 1-line exprs instead of 2-lines ones.